static/src/components/Map/MapPath.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { DEFAULT_PATH_STROKE_WIDTH, DEFAULT_PATH_STROKE_OPACITY } from '../../constants';
+
+const propTypes = {
+  path: React.PropTypes.object,
+  sounds: React.PropTypes.object,
+};
+
+class MapPath extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    // TODO: optimize this
+    return true;
+  }
+
+  render() {
+    return (<g>
```

Comment:
before returning I would consider adding this:
```
if (!this.props.path.sounds.length) {
  return null;
}
```

More readable than the ternary `[...Array((this.props.path.sounds.length) ? this.props.path.sounds.length - 1 : 0).keys()]`